app/controllers/valets_controller.rb

```diff
@@ -0,0 +1,34 @@
+class ValetsController < ApplicationController
+
+  def update
+    @valet = Valet.find_by_id(params[:id])
+    if @valet.nil?
+      render json: {message: "404 Not Found"}
+    else
+      @valet.update(valet_params)
+    end
+  end
+
+  def show
+    @valet = Valet.find_by_id(params[:id])
+    if @valet.nil?
+      render json: {message: "400 Bad Request"}
+    end
+  end
+
+  def index
+    @valets = Valet.all
+  end
+
+  def available
+    @valets = Valet.where(status: "available")
```

Comment:
here you can use `render :index`. Since both your templates are the same, you can just send the `@valets` instance variable to the view and DRY it up :)